docs/applications/consuming-android-libraries.md

```diff
@@ -0,0 +1,196 @@
+---
+id: consuming-android-libraries
+title: Consuming Android libraries
+sidebar_label: Consuming Android libraries
+---
+
+# Consuming Android libraries
+
+## Megazord (composite) libraries
+
+The Rust component libraries that Application Services publishes stand alone: each published Android
+ARchive (AAR) contains managed code (`classes.jar`) and multiple `.so` library files (one for each
+supported architecture).  That means consuming multiple such libraries entails at least two `.so`
+libraries, and each of those libraries includes the entire Rust standard library as well as
+(potentially many) duplicated dependencies.  To save space and allow cross-component native-code
+Link Time Optimization (LTO, i.e., inlining, dead code elimination, etc) Application Services also
+publishes composite libraries -- so called *megazord libraries* -- that compose multiple Rust
+components into a single optimized `.so` library file.  The managed code can be easily configured to
+use such a megazord without significant changes.
+
+There are two tasks we want to arrange.  First, we want to substitute component modules for the single
+megazord module; second, we want to arrange for native Rust code to be available to JVM unit tests.
+
+Both tasks are handled by the [megazord-gradle](https://github.com/mozilla/megazord-gradle) Gradle
+plugin; see that page for details on how to apply the plugin to your build configuration.
+
+## Consuming megazords
+
+To consume a specific megazord module, use something like:
+
+```groovy
+appservices {
+    defaultConfig {
+        // Megazord in all variants.  The default is to not megazord.
+        megazord = 'org.mozilla.appservices.megazord'
+        // With no configuration, the default is to enable unit tests for all variants.
+        enableUnitTests = true
+    }
+
+    // or:
+
+    buildTypes {
+        debug {
+            megazord = 'org.mozilla.appservices.megazord'
+            enableUnitTests = true
+        }
+        // Do not megazord or enable unit tests in release build type.
```

Comment:
Is not enabling it in release builds something we want to encourage, or discourage?  Seems like the later.

Reply:
Correct.  I'll try to clarify this.